src/engine/guidance/collapse_turns.cpp

```diff
@@ -371,6 +371,9 @@ RouteSteps collapseTurnInstructions(RouteSteps steps)
             break;
 
         const auto previous_step = findPreviousTurn(current_step);
+        const auto previous_step_name = facade.GetNameForID(previous_step->name_id).to_string();
+        const auto current_step_name = facade.GetNameForID(current_step->name_id).to_string();
+        const auto next_step_name = facade.GetNameForID(next_step->name_id).to_string();
```

Comment:
Do we really need a string copy here or is a string view good enough?

(the `.to_string()` makes a copy of the internal memory used to represent a string)

Reply:
this can be removed since way names are already available through RouteSteps